token/cli/src/main.rs

```diff
@@ -1891,6 +1911,164 @@ async fn command_sync_native(
     })
 }
 
+// Both enable-required_transfer-mesos and disable-required_transfer-mesos, switches with enable bool
+async fn command_required_transfer_memos(
+    token_account_address: Pubkey,
+    owner: Pubkey,
+    bulk_signers: BulkSigners,
+    enable: bool,
+    config: &Config<'_>,
+) -> CommandResult {
+    let program_id = if config.sign_only {
+        config.program_id
+    } else {
+        config
+            .rpc_client
+            .get_account(&token_account_address)
+            .await
+            .map_err(|err| {
+                format!(
+                    "Token account {} does not exist: {}",
+                    token_account_address, err
+                )
+            })?
+            .owner
+    };
+    let mut instructions: Vec<Instruction> = Vec::new();
+    let reallocate_instruction = reallocate_if_needed(
+        &program_id,
+        &token_account_address,
+        &owner,
+        vec![ExtensionType::MemoTransfer],
+        config,
+    )
+    .await?;
+    if let Some(instr) = reallocate_instruction {
+        instructions.push(instr);
+    }
+    if enable {
+        instructions.push(
+            spl_token_2022::extension::memo_transfer::instruction::enable_required_transfer_memos(
+                &program_id,
+                &token_account_address,
+                &owner,
+                &config.multisigner_pubkeys,
+            )?,
+        );
+    } else {
+        instructions.push(
+            spl_token_2022::extension::memo_transfer::instruction::disable_required_transfer_memos(
+                &program_id,
+                &token_account_address,
+                &owner,
+                &config.multisigner_pubkeys,
+            )?,
+        );
+    }
+    let tx_return = handle_tx(
```

Comment:
note to jon when i tag him in that im merging extension implementations that use this as-is, and will do a second batch of token client conversions later